extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/TranslatableInvalidHostLoginException.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.guacamole.auth.restrict;
+
+import org.apache.guacamole.language.TranslatableGuacamoleClientException;
+import org.apache.guacamole.language.TranslatableMessage;
+
+/**
+ * An exception that represents an invalid login or connection due to
+ * restrictions based on the host from which the action should be allowed.
+ */
+public class TranslatableInvalidHostLoginException 
+        extends TranslatableGuacamoleClientException {
+
+    /**
+     * The serial version ID of this class.
+     */
+    private static final long serialVersionUID = 1L;
```

Comment:
Good call on adding these - I had forgotten that `Exception` is `Serializable`, so this seems like the right thing to do. That said, do we actually serialize any `GuacamoleException`s? We have a lot of subclasses floating around without this field being defined...

Reply:
I'm not sure - NetBeans recommended it, so I added it. I'm happy to go either way, although it's unclear to me what the value is if the superclasses don't also have it. I'm also happy to go back and add it to others, if we should do that.